### ISSUE TITLE
Snakemake incompatible with Python 3.11 due to `ratelimiter` dependency

### ISSUE THREAD

**Issue by pvandyken** [snakemake/snakemake#1952]
**Snakemake version**

Any version

**Describe the bug**
Snakemake uses the `ratelimiter` library to manage rate-limited api calls (mainly for checking if remote jobs are finished). This library uses a deprecated method on `asyncio`: `asyncio.coroutine`, that has been fully removed in Python 3.11. As a result, Snakemake will not run at all on Python 3.11

**Logs**
```
Traceback (most recent call last):
  File "/home/runner/work/snakebids/snakebids/.venv/lib/python3.11/site-packages/snakemake/__init__.py", line 747, in snakemake
    success = workflow.execute(
              ^^^^^^^^^^^^^^^^^
  File "/home/runner/work/snakebids/snakebids/.venv/lib/python3.11/site-packages/snakemake/workflow.py", line 949, in execute
    self.scheduler = JobScheduler(
                     ^^^^^^^^^^^^^
  File "/home/runner/work/snakebids/snakebids/.venv/lib/python3.11/site-packages/snakemake/scheduler.py", line 109, in __init__
    from ratelimiter import RateLimiter
  File "/home/runner/work/snakebids/snakebids/.venv/lib/python3.11/site-packages/ratelimiter.py", line 36, in <module>
    class RateLimiter(object):
  File "/home/runner/work/snakebids/snakebids/.venv/lib/python3.11/site-packages/ratelimiter.py", line 127, in RateLimiter
    __aexit__ = asyncio.coroutine(__exit__)
```

**Minimal example**
Any snakemake call will trigger this

**Additional context**
There's another library, [throttle](https://github.com/uburuntu/throttler), that should be a drop-in replacement for ratelimiter and doesn't have the same problems.

**Comment by hermidalc** [comment]
I can reproduce this error as well and see it in my environment

**Comment by MilesCranmer** [comment]
Also seeing this error on Python 3.11.

You can patch this locally with this PR: https://github.com/RazerM/ratelimiter/pull/11 applied to the `ratelimiter.py` file. This error fixes it for me.

However, it looks like the repository hasn't had a patch since 2018 so it would be preferred if @pvandyken's suggested package was used instead.

**Comment by johanneskoester** [comment]
The issue is fixed for now when using conda, as the ratelimiter package has been patched to constrain itself to Python <3.11. However, this constraint is not visible for pip, as we cannot modify the package over there. This nicely illustrates that it is a good idea to keep package maintenance separate from software authorship, and in particular in the hands of the community.
